### DISCUSSION TITLE
"inactive" option for refetchOnMount

### DISCUSSION THREAD

**Discussion by BenLorantfy** [TanStack/query#1991] [Ideas]
As of today `refetchOnMount` has these options:
```
refetchOnMount: boolean | "always"
- Optional
- Defaults to true
- If set to true, the query will refetch on mount if the data is stale.
- If set to false, the query will not refetch on mount.
- If set to "always", the query will always refetch on mount.
```

I'd like to propose adding an "inactive" option that only refetches on mount if the query was previously inactive before mount.  I'd like to try working on this feature if it's something that is desired.

EDIT
I created this commit to help discussion as it basically does what I'm thinking: https://github.com/BenLorantfy/react-query/commit/9e1f6a9f03c656f346d626c451b94014b8b7295f .  I can create a proper PR later if wanted.

**Comment by BenLorantfy** [comment]
@TkDodo would you be open for a PR for this?

**Comment by TkDodo** [comment]
I'd like to understand the use-case for something like this please.

**Reply by BenLorantfy** [reply]
The use-case is to minimize the number of requests.  If the data is already being viewed on the page, then I'd like to avoid making another request for what is likely the same data.  In an ideal world it makes sense to eagerly fetch the data, but not all backends are scalable enough to support this type of data-fetching.  

Another use-case: if the data has already been loaded, it could be disruptive/surprising to the user for new data to be fetched.  Some users assume the data remains the same while they are on the page.  Some users are trained to refresh the page if they want new data.

I'm mostly interested in the first use case though.  I could set this option to "false" but then in a SPA the user would navigate around and never get new data.  If the user leaves the page and comes back I'd like it to fetch new data.

**Reply by TkDodo** [reply]
Isn't this what `staleTime` is usually for? With `staleTime`, you can customize how long data is going to be `fresh`, and as long as it is, no background update will happen - data will always come from the cache. So if you know that your data will be good for 3 minutes, you can set `staleTime` to `3 * 60 * 1000`.

The idea behind it is that caching is usually time based. For example, the http cache header `Cache-Control: max-age` works in a similar way. Having something mounted already usually does not indicate that it is unnecessary to refetch data. Suppose you log in and show some data, then go on lunch break and come back 20 minutes later. Just because you have data mounted on the screen doesn't mean it's unnecessary to refetch it once a new observer mounts now. I think `staleTime` solves this in an elegant way.
